### PR TITLE
fix: retry timeout of 60s for reconnection

### PR DESCRIPTION
**Pull request by eswarclynn**
### Details(context, link the issue, how was the bug fixed, what does the new feature do)

-
-

### CODE REVIEW

File: packages/hms-video-store/src/transport/RetryScheduler.ts [hunk 1]
```diff
@@ -171,33 +173,37 @@ export class RetryScheduler {
...
+    const jitter = category === TFC.JoinWSMessageFailed ? Math.random() * 2 : Math.random();
+    let delaySeconds = 0;
     if (category === TFC.JoinWSMessageFailed) {
       // linear backoff(2 + jitter for every retry)
-      return 2;
+      delaySeconds = 2 + jitter;
+    } else if (category === TFC.SignalDisconnect) {
+      delaySeconds = 1;
+    } else {
+      delaySeconds = 0;
```
Comment (raviteja83) [inline packages/hms-video-store/src/transport/RetryScheduler.ts:203]: 0? retry immediately? for which cases? this else should be covered with initialisation.

File: packages/hms-video-store/src/transport/RetryScheduler.ts [hunk 2]
```diff
@@ -171,33 +173,37 @@ export class RetryScheduler {
...
-  private getBaseDelayForTask(category: TFC, n: number) {
+  private getDelayForRetryCount(category: TFC) {
+    const jitter = category === TFC.JoinWSMessageFailed ? Math.random() * 2 : Math.random();
+    let delaySeconds = 0;
     if (category === TFC.JoinWSMessageFailed) {
       // linear backoff(2 + jitter for every retry)
-      return 2;
+      delaySeconds = 2 + jitter;
+    } else if (category === TFC.SignalDisconnect) {
+      delaySeconds = 1;
```
Comment (raviteja83) [inline packages/hms-video-store/src/transport/RetryScheduler.ts:201]: should we keep it as 2?

File: packages/hms-video-store/src/transport/RetryScheduler.ts [hunk 3]
```diff
@@ -65,9 +65,10 @@ export class RetryScheduler {
     changeState,
     task,
     originalState,
-    maxFailedRetries = MAX_TRANSPORT_RETRIES,
+    failedAt,
+    maxRetryTime = MAX_TRANSPORT_RETRY_TIME,
     failedRetryCount = 0,
-  }: ScheduleTaskParams & { failedRetryCount?: number }): Promise<void> {
+  }: ScheduleTaskParams & { failedAt: number; failedRetryCount?: number }): Promise<void> {
```
Comment (raviteja83) [inline packages/hms-video-store/src/transport/RetryScheduler.ts:71]: is failedRetryCount still needed?

Comment (eswarclynn) [reply packages/hms-video-store/src/transport/RetryScheduler.ts:71]: failedRetryCount is still needed to figure out if it's the first retry to send an update to the client

Android also has a isFirstRetry flag param to notify this

File: packages/hms-video-store/src/transport/RetryScheduler.ts (hunk 1 above)
Comment (eswarclynn) [reply packages/hms-video-store/src/transport/RetryScheduler.ts:203]: ConnectFailed(init, opening websocket), ice failed

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**